src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -262,6 +262,16 @@ private ConstantValue DefaultSyntaxValue
                         }
                     }
 
+                    // Check availability of `DecimalConstantAttribute`.
+                    if (_lazyDefaultSyntaxValue is { IsDecimal: true } ||
+                        GetEarlyDecodedWellKnownAttributeData() is { DefaultParameterValue.IsDecimal: true })
```

Comment:
>GetEarlyDecodedWellKnownAttributeData() is { DefaultParameterValue.IsDecimal: true } [](http://example.com/codeflow?start=24&length=84)

I am not sure what are we trying to protect against with this condition. Explicit attribute applications are either bound successfully, or not.